synapse/daemon.py

```diff
@@ -433,19 +479,22 @@ def _onTelePushMesg(self, sock, mesg):
 
         jid = mesg[1].get('jid')
         name = mesg[1].get('name')
+        csides = mesg[1].get('csides')
         reflect = mesg[1].get('reflect')
 
         user = sock.get('syn:user')
         if not self._isUserAllowed(user, 'tele:push:'+name ):
             return sock.tx( tufo('job:done', err='NoSuchRule', jid=jid) )
 
         def onfini():
-            self.pushed.pop(name, None)
-            self.reflect.pop(name, None)
+            self.fire('tele:push:fini', name=name)
```

Comment:
Dont see too much difference either way.  Did you have a scenario in mind i wasn't thinking of?